### PR TITLE
Spinning fidget spinners

### PR DESCRIPTION
**Pull request by The-Anjelo-Guy**
#### Summary
Content "Add item action 'SPIN' for spinning fidget spinners"

#### Purpose of change

Just for fluff purposes.

#### Describe the solution

Not doing it.

### CODE REVIEW

**Comment by GuardianDll** [comment]
It is not how it works, and it's not how you need to make it work

**Comment by The-Anjelo-Guy** [comment]
> It is not how it works, and it's not how you need to make it work

Added item action, spin, added action to item, that is, spin. Any C++ I should've know? If so, tell me.

**Comment by GuardianDll** [comment]
If all you want is print some message when item is activated, it is handled by a simple EoC action

**Comment by The-Anjelo-Guy** [comment]
> If all you want is print some message when item is activated, it is handled by a simple EoC action

Probably some seconds where the player character is spinning the fidget spinner, a waiting message saying "spinning, press (any button) to stop", and a message saying "you spin your fidget spinner."

**Comment by The-Anjelo-Guy** [comment]
> With how fast our game process, you could spin it for hours in a second Don't do this, that's silly

Alright then, no time. But I do want the message "You spin your fidget spinner".

**Comment by Brambor** [comment]
I recently saw this
"Don't Contribute to Open Source" https://www.youtube.com/watch?v=5nY_cy8zcO4
This PR reminds me of it.

If I saw option SPIN on a fidget spinner, I would assume it does _something_, like giving a morale boost. When you play an instrument, it gives you morale by listening to music.

btw you have:
> Not doing it.

in
> Describe the solution

instead of
> Describe alternatives you've considered

I just don't get the point.
Is it a funny contribution? Maybe.
Does it contribute to the game? How? If not, don't do it.


Why not add "click" to pens "tap" to sticks and mugs?

**Comment by GuardianDll** [comment]
> Not doing it. 

I think it was just a mistake, and intention is to do it, and not doing it is alternative 

> Does it contribute to the game?

Not now, but when it would be finished, it would be, at first, another example of EoC inlining to make simple effect, and second, why not? In case of EoC it would be a small, few line EoC, inlined inside the item, not scratched across bunch of files, and people like this little silly things

> Why not add "click" to pens "tap" to sticks and mugs?

Again, what's the problem? Not every thing in the game should serve The Great Purpose nor be useful on its own. Adding mood effect would make a bunch of players shot themselves into the foot and use it to handle bad focus. It may make laugh few players, and it cause no another harm nor difficulties, and it's enough reason to finish it and merge it

File: data/json/item_actions.json [hunk 1]
```diff
@@ -540,6 +540,11 @@
     "id": "PLAY_GAME",
     "name": { "str": "Play" }
   },
+  {
+    "type": "item_action",
+    "id": "SPIN",
+    "name": { "str": "Spin" }
+  },
```
Comment (GuardianDll) [inline data/json/item_actions.json:547]:
```suggestion
```

File: data/json/items/toy.json [hunk 2]
```diff
@@ -884,6 +884,7 @@
     "weight": "75 g",
     "volume": "185 ml",
     "longest_side": "10 cm",
+    "use_action": [ "SPIN" ],
```
Comment (GuardianDll) [inline data/json/items/toy.json:887]:
```suggestion
    "use_action": {
      "type": "effect_on_conditions",
      "menu_text": "Spin",
      "effect_on_conditions": [
        {
          "id": "EOC_spinner_spinning",
          "effect": { "u_message": "You spin your fidget spinner." }
        }
      ]
    },
```

**Comment by The-Anjelo-Guy** [comment]
> I recently saw this "Don't Contribute to Open Source" https://www.youtube.com/watch?v=5nY_cy8zcO4 This PR reminds me of it.
> 
> If I saw option SPIN on a fidget spinner, I would assume it does _something_, like giving a morale boost. When you play an instrument, it gives you morale by listening to music.
> 
> btw you have:
> 
> > Not doing it.
> 
> in
> 
> > Describe the solution
> 
> instead of
> 
> > Describe alternatives you've considered
> 
> I just don't get the point. Is it a funny contribution? Maybe. Does it contribute to the game? How? If not, don't do it.
> 
> Why not add "click" to pens "tap" to sticks and mugs?

Those last two are great ideas. Thank you so much for your support.

**Comment by Brambor** [comment]
I encountered some of your other PRs.

I feel like you are using Chat GPT to generate content to add to the game.

**Comment by The-Anjelo-Guy** [comment]
> I encountered some of your other PRs.
> 
> I feel like you are using Chat GPT to generate content to add to the game.

Sorry English is not my first language so I have no idea what you are talking about, lol.

**Comment by Fris0uman** [comment]
Does this add an entry to a menu? It'd be better if it didn't, it's fine if it's an easter egg type thing that's not obvious to find, since it's pure flavor.

**Comment by GuardianDll** [comment]
There is no way to add use action to item without it being shown in the action menu

**Pull request merged**